modules/aws-emr-ports/output.tf

```
@@ -0,0 +1,21 @@
+locals {
+  applications  = [for app in var.applications : lower(app)]
+  running_hbase = contains(local.applications, "hbase")
+  running_spark = contains(local.applications, "spark")
+  output_hbase  = concat(var.master_ports_emr, var.master_ports_hbase, var.master_ports_ganglia, var.additional_ports)
+  output_spark  = concat(var.master_ports_emr, var.master_ports_spark, var.additional_ports)
+  output        = concat(var.master_ports_emr, var.additional_ports)
+}
+
+output "ingress_master_ports" {
+  value = local.running_hbase ? local.output_hbase : (local.running_spark ? local.output_spark : local.output)
```

We could probably simplify this a bit.

If we did something like this in our locals
```
  output_spark_master  = running_spark ? var.master_ports_spark : []
```

Then this could be the composite.
```suggestion
  value = flatten(local.output_spark_master, local.output_hbase_master, local.output_ganglia_master, var.additional_ports_master)
```

---

Done